### ISSUE TITLE
Allow payments from unconfirmed push_amount

### ISSUE THREAD

**Issue by Kixunil** [lightningnetwork/lightning-rfc#565]
When push amount is used, waiting for confirmation is not necessary to spend it, since in the worst case, the transaction doesn't confirm and becomes fully "spent from" anyway.

This should be part of specification, to achieve better interoperability.

Related issue: https://github.com/btcontract/lnwallet/issues/125

**Comment by btcontract** [comment]
I've implemented a prototype in testnet version of BLW, next I'll implement this in a fork of Eclair, here's how it works thus far:

1. Defined a `turbo` feature bit in `OpenChannel.channelFlags` as `(channelFlags & 0x04) != 0` (since `0x02` and `0x03` are for wumbo and simplified commitment?).
2. On funder/fundee sides: if `turbo` is defined, then both peers exchange `FundingLocked` and enter operational channel state as soon as they broadcast a funding (funder) / see funding in a mempool (fundee).

Feature bit naming as well as general suggestions/criticism are very welcome!

**Comment by Kixunil** [comment]
Nice, I'm going to try it out!

Regarding naming, I think `turbo` is cute, but I prefer more meaningful names for things. I'd suggest `zeroconf_spendable_push`

**Comment by molxyz** [comment]
Dangerous proposal! If you want zero conf, use bcash!!!!!  This would be a bad precedent if introduced to LN; however, nobody would stop you from wanting to be robbed though, can't you implement it only for your own biz?  👅 

NACK!

**Comment by Kixunil** [comment]
Please read the whole proposal again first. This specific case is safe.

Here's a detailed blog post about it: https://medium.com/@akumaigorodski/instant-channels-enable-safe-lightning-payments-with-unconfirmed-funding-8d640defa183

Writing such claims and thumbsdowning everything only trashes your reputation.

**Comment by btcontract** [comment]
@molxyz your critique would only be valid if you never ever used an exchange or purchased bitcoin with fiat, otherwise it's misguided.

**Comment by Kixunil** [comment]
@btcontract heh, not even that. He'd have to never get bitcoin trusting other party - no work for bitcoins, no donations. Only mining would be a valid argument.

**Comment by btcontract** [comment]
@Kixunil 
> Regarding naming, I think turbo is cute, but I prefer more meaningful names for things. I'd suggest zeroconf_spendable_push

Makes sense as this is indeed very specific, I've renamed it in both BLW and Eclair fork.

**Comment by Kixunil** [comment]
What we can do to get this moving? Reference implementation is there, the bit is defined. (Although, @btcontract, here you say it's 0x04, but in the code I see 0x08, could you clarify please?)

What more is needed?
